public/app/percona/add-instance/components/AddRemoteInstance/FormParts/ExternalServiceConnectionDetails/ExternalServiceConnectionDetails.tsx

```diff
@@ -1,17 +1,19 @@
-import React, { FC, useCallback, useMemo } from 'react';
+import React, { FC, useCallback, useEffect, useMemo } from 'react';
 import { PasswordInputField, RadioButtonGroupField, TextInputField, validators } from '@percona/platform-core';
 import Validators from 'app/percona/shared/helpers/validators';
 import { LinkTooltip } from 'app/percona/shared/components/Elements/LinkTooltip/LinkTooltip';
-import { Button, useTheme } from '@grafana/ui';
-import { ExternalServiceConnectionDetailsFormPartProps } from '../FormParts.types';
+import { useTheme } from '@grafana/ui';
+import { ExternalServiceConnectionDetailsFormPartProps, MetricsParameters, Schema } from '../FormParts.types';
 import { getStyles } from '../FormParts.styles';
 import { Messages } from '../FormParts.messages';
-import { schemaOptions } from '../FormParts.constants';
+import { metricsParametersOptions, schemaOptions } from '../FormParts.constants';
 
 export const ExternalServiceConnectionDetails: FC<ExternalServiceConnectionDetailsFormPartProps> = ({ form }) => {
   const theme = useTheme();
   const styles = getStyles(theme);
 
+  const selectedOption = form.getState().values && form.getState().values['metricsParameters'];
```

Comment:
We can assign `form.getState()` to a variable and use that variable instead to avoid getting the state multiple times.

Reply:
Tiago, thank you! Fixed it